app/components/footer.tsx

```diff
@@ -0,0 +1,202 @@
+}
+
+function FooterLink({name, href}: FooterLinkProps) {
+  return (
+    <li>
+      <a
+        href={href}
+        className="inline-block py-1 dark:text-blueGray-500 text-gray-500 hover:underline whitespace-nowrap text-lg"
+      >
+        {name}
+      </a>
+    </li>
+  )
+}
+
+function GithubLink() {
+  return (
+    <a href="https://kcd.im/github">
+      <svg className="w-8 h-8" viewBox="0 0 24 24">
+        <path
+          fill="currentColor"
+          d="M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z"
+        />
+      </svg>
+    </a>
+  )
+}
+
+function YoutubeLink() {
+  return (
+    <a href="https://kcd.im/twitter">
+      <svg className="w-8 h-8" viewBox="0 0 24 24">
+        <path
+          fill="currentColor"
+          d="M10,15L15.19,12L10,9V15M21.56,7.17C21.69,7.64 21.78,8.27 21.84,9.07C21.91,9.87 21.94,10.56 21.94,11.16L22,12C22,14.19 21.84,15.8 21.56,16.83C21.31,17.73 20.73,18.31 19.83,18.56C19.36,18.69 18.5,18.78 17.18,18.84C15.88,18.91 14.69,18.94 13.59,18.94L12,19C7.81,19 5.2,18.84 4.17,18.56C3.27,18.31 2.69,17.73 2.44,16.83C2.31,16.36 2.22,15.73 2.16,14.93C2.09,14.13 2.06,13.44 2.06,12.84L2,12C2,9.81 2.16,8.2 2.44,7.17C2.69,6.27 3.27,5.69 4.17,5.44C4.64,5.31 5.5,5.22 6.82,5.16C8.12,5.09 9.31,5.06 10.41,5.06L12,5C16.19,5 18.8,5.16 19.83,5.44C20.73,5.69 21.31,6.27 21.56,7.17Z"
+        />
+      </svg>
+    </a>
+  )
+}
+
+function TwitterLink() {
+  return (
+    <a href="https://kcd.im/twitter">
+      <svg className="w-8 h-8" viewBox="0 0 24 24">
+        <path
+          fill="currentColor"
+          d="M22.46,6C21.69,6.35 20.86,6.58 20,6.69C20.88,6.16 21.56,5.32 21.88,4.31C21.05,4.81 20.13,5.16 19.16,5.36C18.37,4.5 17.26,4 16,4C13.65,4 11.73,5.92 11.73,8.29C11.73,8.63 11.77,8.96 11.84,9.27C8.28,9.09 5.11,7.38 3,4.79C2.63,5.42 2.42,6.16 2.42,6.94C2.42,8.43 3.17,9.75 4.33,10.5C3.62,10.5 2.96,10.3 2.38,10C2.38,10 2.38,10 2.38,10.03C2.38,12.11 3.86,13.85 5.82,14.24C5.46,14.34 5.08,14.39 4.69,14.39C4.42,14.39 4.15,14.36 3.89,14.31C4.43,16 6,17.26 7.89,17.29C6.43,18.45 4.58,19.13 2.56,19.13C2.22,19.13 1.88,19.11 1.54,19.07C3.44,20.29 5.7,21 8.12,21C16,21 20.33,14.46 20.33,8.79C20.33,8.6 20.33,8.42 20.32,8.23C21.16,7.63 21.88,6.87 22.46,6Z"
+        />
+      </svg>
+    </a>
+  )
+}
+
+// mt-20 md:mt-36
+function Footer() {
```

Comment:
@kentcdodds, tabbing in the footer component feels a bit counterintuitive on `dekstop` resolutions. Elements are rearranged when resizing the browser, but <kbd>tab</kbd> follows the DOM, and not the visual order. So on desktop, focus jumps from the social icons on the left, to the newsletter signup on the right, to the nav links in the center.

On smaller resolution, the focus follows the right path. I'm not sure what's the right thing to do here. I see two solutions:

1. Add javascript width detection, and enforce tab order that way (not a fan)
2. Duplicate the markup for the newsletter, hide the version out of range, and show the version in range. That means double HTML, but the focus works as expected. (not really awesome either)

Do you have other ideas/suggestions?

Reply:
That should work. That's what I meant with "2". Let's do that.